gcloud-java-bigquery/src/main/java/com/google/gcloud/spi/DefaultBigQueryRpc.java

```diff
@@ -0,0 +1,421 @@
+import com.google.api.services.bigquery.model.Dataset;
+import com.google.api.services.bigquery.model.DatasetList;
+import com.google.api.services.bigquery.model.DatasetReference;
+import com.google.api.services.bigquery.model.GetQueryResultsResponse;
+import com.google.api.services.bigquery.model.Job;
+import com.google.api.services.bigquery.model.JobList;
+import com.google.api.services.bigquery.model.JobReference;
+import com.google.api.services.bigquery.model.QueryRequest;
+import com.google.api.services.bigquery.model.QueryResponse;
+import com.google.api.services.bigquery.model.Table;
+import com.google.api.services.bigquery.model.TableDataInsertAllRequest;
+import com.google.api.services.bigquery.model.TableDataInsertAllResponse;
+import com.google.api.services.bigquery.model.TableDataList;
+import com.google.api.services.bigquery.model.TableList;
+import com.google.api.services.bigquery.model.TableReference;
+import com.google.api.services.bigquery.model.TableRow;
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+
+import static com.google.gcloud.spi.BigQueryRpc.Option.MAX_RESULTS;
+import static com.google.gcloud.spi.BigQueryRpc.Option.PAGE_TOKEN;
+
+import com.google.gcloud.bigquery.BigQueryException;
+import com.google.gcloud.bigquery.BigQueryOptions;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class DefaultBigQueryRpc implements BigQueryRpc {
+
+  public static final String DEFAULT_PROJECTION = "full";
+  // see: https://cloud.google.com/bigquery/troubleshooting-errors
+  private static final Set<Integer> RETRYABLE_CODES = ImmutableSet.of(503);
+  private final BigQueryOptions options;
+  private final Bigquery bigquery;
+
+  public DefaultBigQueryRpc(BigQueryOptions options) {
+    HttpTransport transport = options.httpTransportFactory().create();
+    HttpRequestInitializer initializer = options.httpRequestInitializer();
+    this.options = options;
+    bigquery = new Bigquery.Builder(transport, new JacksonFactory(), initializer)
+        .setRootUrl(options.host())
+        .setApplicationName(options.applicationName())
+        .build();
+  }
+
+  private static BigQueryException translate(IOException exception) {
+    BigQueryException translated;
+    if (exception instanceof GoogleJsonResponseException) {
+      translated = translate(((GoogleJsonResponseException) exception).getDetails());
+    } else {
+      translated =
+          new BigQueryException(BigQueryException.UNKNOWN_CODE, exception.getMessage(), false);
+    }
+    translated.initCause(exception);
+    return translated;
+  }
+
+  private static BigQueryException translate(GoogleJsonError exception) {
+    boolean retryable = RETRYABLE_CODES.contains(exception.getCode())
+        || "InternalError".equals(exception.getMessage());
```

Comment:
I think this line should be `|| "internalError".equals(exception.getReason());` because [this link](https://cloud.google.com/bigquery/troubleshooting-errors) (second line under "Error table") says the "error code" (which looks like it's camel case) maps to the error's reason.